### DISCUSSION TITLE
Weekly Blog 3

### DISCUSSION THREAD

**Discussion by dolerich-hirnfiedler** [SE-TINF22B6/Underwatch#36] [Announcements]
Hello and welcome to the third edition of our weekly blog for the Underwatch Game. This week we will have a glance at our awesome updated documentation. Furthermore, we will have a look at the progress made in the project itself.

# 📑 Documentation
We use the [Arc42](https://arc42.org/) documentation template for our documentation. We encourage you to check it out for yourself, as it is a widely used industry standard.
The update to our documentation entails three points:

1. We added [Personas](https://github.com/SE-TINF22B6/Underwatch/wiki/System-Architecture-Documentation#131-personas) for our users. 
2. We added diagrams for [functional use cases](https://github.com/SE-TINF22B6/Underwatch/wiki/System-Architecture-Documentation#31-business-context)
3. We added [sequence and activity diagrams](https://github.com/SE-TINF22B6/Underwatch/wiki/System-Architecture-Documentation#6-runtime-view) for some of the use-cases.

This is the functional usecase for the final game:
![ucd_001](https://github.com/SE-TINF22B6/Underwatch/assets/87950162/51ce04d9-f5cb-478e-ba93-aef7bbfb6128)

# 🌐 Website
On the website, we included routing to the different pages presented in our [last post](https://github.com/SE-TINF22B6/Underwatch/discussions/22). We also included a custom 404 Page. Currently, these pages are mere placeholders. In the future, we will have exciting pages to show.

# 🍃 Backend
In the Backend we decided to use [Spring Boot](https://spring.io/) as our framework. We also decided to use [postgresql](https://www.postgresql.org/) for our database. In order to start the backend development soon, we set up a docker configuration for local development via [docker-compose](https://docs.docker.com/compose/). We also included two configurations for our spring-boot backend. One for development, called dev the other one for production, called prod. The corresponding files can be found in our github-repository:

1. [docker-compose file](https://github.com/SE-TINF22B6/Underwatch/blob/main/backend/docker-compose.yml)
2. [spring-boot configuration](https://github.com/SE-TINF22B6/Underwatch/tree/main/backend/src/main/resources)

**Comment by derrrobin** [comment]
Hello all,

We as CookPit team are very enthusiastic about your reports. We like them structurally very much and have a consistent line. I like that you talked about your ideas and designs in the last weeks and this week you talk more about the technologies and the implementation.  Also your diagram which you have uploaded is very structured and good for a lender, because he understands your diagram on anhib and it was not too complex gestalltet.

Keep up the good work.

Your CookPit Team

**Comment by giomskii** [comment]
Your documentation is very detailed, especially the huge amount of activity and sequence diagrams picture a good outline of a possible finished product. It's also interesting, that you additionally have 3 distinct personas that the game needs to appeal to. I think it would have been nice to include some of the examples on your srs in this blog so that we directly can see them. 👍

**Comment by fehlend0** [comment]
Hello dear Underwatch team,

adding personas for your users is a smart move, as it helps to humanize your project and consider the end-users' needs. The inclusion of routing to different pages and a custom 404 Page on your website is also a good step. While the current pages are placeholders, it's exciting to hear that you have plans for more engaging content in the future. As for the tools you use, selecting Spring Boot as your framework and PostgreSQL for your database is a solid choice. The setup of a docker configuration for local development using docker-compose is a good practical decision. And last but not least, the inclusion of configurations for both development  and production in your GitHub repository is good organized and helpful for the future navigation.

Overall, your project is progressing well, and it's evident that you are taking important steps to document, design, and develop efficiently. Keep up the good work, and we look forward to hearing more about Underwatch in the future!

Your Statify Team

**Comment by AndresMasis** [comment]
Hi. First I really appreciate that you suggested the Arc42 standard. I did not know it, so it is always great to learn from other teams. I also want to confratulate you by your work because it seems very well structured.  I think that the point of the Personas is very important for the development of the product because it makes it easier to understand the needs of an actual stakeholder.

Cheers from CookPit.
